oidc-wp/openid-connect-generic
# Add openid-connect-generic-session-expired action on session expiry before logout

Add openid-connect-generic-session-expired action before to allow consumers to act on the refresh token expiration

### All Submissions:






### Changes proposed in this Pull Request:



Added action 'openid-connect-generic-session-expired' to allow client to react to the session expiry (refresh token expired) just before logging the user out and redirecting them back to the login page.    It can be useful in situations where SSO is being used across multiple apps.  In this case the user is logged in with the SSO provider via another app, but has timed out of wordpress.  This allows a hook to re-authenticate them automatically and allow access to the resource based.  The action provides the user and the REDIRECT_URI as parameters to allow any necessary action to be taken on timeout with the url and parameters provided.   

Closes # .

### How to test the changes in this Pull Request:

1. Add the following hook in the client code
    add_action('openid-connect-generic-session-expired', 'expired_method', 10, 2);
2.  Start a session with the OpenID Connect provider via wordpress.   Close the browser without logging out and wait until the token refresh has expired.  
3.  Revisit the wordpress resource and observe the action being fired. 

### Other information:




### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.
